skt/__init__.py

```diff
@@ -493,8 +497,25 @@ def getrelease(self):
 
         return krelease
 
-    def mktgz(self, clean=True):
-        tgzpath = None
+    def mktgz(self, clean=True, timeout=60 * 60 * 12):
+        """
+        Build kernel and modules, after that, pack everything into a tarball.
+
+        Args:
+            clean:      If it is True, run the `mrproper` target before build.
+            timeout:    Max time in seconds will wait for build.
+        Returns:
+            The full path of the tarball generated.
+        Raises:
+            CommandTimeoutError: When building kernel takes longer than the
+                                 specified timeout.
+            CalledProcessError:  When a command returns an exit code different
```

Comment:
Shouldn't this say "When "make" returns an exit code..."? Or do you want to keep that abstracted away, or plan adding other commands to this?

Reply:
Yes, I'd like to keep it abstracted because it calls to `.prepare` which uses check_call and the latter raises this exception is something goes wrong.